tests/streams_test.py

```diff
@@ -229,3 +237,85 @@ def test_xc_attainment_measure_criteria(xc_attainment_measure_criteria_obj):
         assert "MODIFIED_DATE" in record
         assert "MODIFIED_BY_ID" in record
         assert "HISTORY_UUID" in record
+
+
+def test_xc_credit(xc_credit_obj):  # pylint: disable=too-many-statements
+    assert xc_credit_obj.tap_stream_id == "xc_credit"
+    assert xc_credit_obj.key_properties == ["CREDIT_ID"]
+    assert xc_credit_obj.object_type == "XC_CREDIT"
+    assert xc_credit_obj.valid_replication_keys == ["MODIFIED_DATE"]
+    assert xc_credit_obj.replication_key == "MODIFIED_DATE"
+
+    assert "xc_credit" in STREAMS
+    assert STREAMS["xc_credit"] == XcCredit
+
+    records = xc_credit_obj.client.query_database(
+        table_name="xc_credit",
+        primary_key="credit_id",
+        limit=5,
+        offset=0,
+        limit_key="MODIFIED_DATE",
+        limit_key_value="2021-05-21T00:52:18",
+    )
+    assert len(records) > 0
```

Comment:
This doesn't test what we need it to, and that's the `sync()` method for this class. That said, it does bring up a point on how we could refactor these tests. We could instead test `Stream.sync()` on one table and then just test the individual properties, i.e. `tap_stream_id`, `key_properties`, etc. and ensure the specific stream object is in `STREAMS`. This should allow the tests to run quicker since we don't need to query the db for every table. It does run the risk of our specific stream properties being incorrect.

Reply:
I apologize I didn't see this comment until now. The assertions are checking the same things whether they are checking on the records form the `sync()` or they are checking from `query_database()`. The Sync method calls the `query_database` method, after all. What is being called in either case is that the length of the records array is more than 0 and then it's checking each record for keys.

I do agree, however, that we should just drop the sync check for each table and only check the stuff before it.

Reply:
This test doesn't exist anymore. Resolving.